kotlinx-coroutines-core/native/src/CoroutineExceptionHandlerImpl.kt

```diff
@@ -5,11 +5,14 @@
 package kotlinx.coroutines
 
 import kotlin.coroutines.*
+import kotlin.system.exitProcess
+import kotlin.native.*
+import kotlin.native.concurrent.*
 
 internal actual fun handleCoroutineExceptionImpl(context: CoroutineContext, exception: Throwable) {
-    // log exception
-//    println("Exception in \"${Worker.current}\"")
-//    exception.printStackTrace()
-// todo: printing exception does not make it easy to debug (no source location), so let it crash instead
-    throw exception
+    // Use unhandled exception hook if it is set, otherwise print the stacktrace
+    val hook = setUnhandledExceptionHook({ _: Throwable -> }.freeze())
+    if (hook != null) hook(exception) else exception.printStackTrace()
```

Comment:
It worth prepending the coroutine context and the fact that an app is terminated due to unhandled exceptions before calling `exitProcess(-1)`

Reply:
Hmmm.... I see that this PR is related to native and I am not concerned at the moment. Also I don't do mobile an don't know the details of setup, so my question might look stupid. But can we be sure that this change is confined to iOS? In other words, are we sure that kotlin native with coroutines will never be used for server side or desktop development? Because `exitProcess(-1)` seems too radical for me on the server, I would definitely hate it. Again sorry for stupid question.

Reply:
Your question raised an important issue that we're going to address, thanks for that!

It is indeed important to have an ability _not_ to terminate an application on unhandled exceptions, but it also important to be able to do so.
KMM and iOS is our primary focus for now and there is only an API to do the latter, not the former.

So, in the `1.5.1-native-mt` the default behaviour will be to crash (in the end, an unhandled exception in a coroutine is no different from the unhandled exception from `main` or a worker) for the sake of iOS apps. The workaround to avoid application crashes is not to leave exceptions unattended (e.g. by using `CoroutinesExceptionHandler`).

But in 1.5.30 (or in 1.6.0, it's being discussed right now) we'll provide a new K/N stdlib API that allows to invoke unhandled exception hook without termination (now this API is supposed to be used _only_ with the consecutive call to `abort` or `exitProcess`) and will provide an opt-in mechanism to set your own exception hook without crashing.